File: Tech Module/Software Technologies/Java/Calculator-SpringMVC/src/main/java/com/softuni/controller/HomeController.java
package com.softuni.controller;

import com.softuni.entity.Calculator;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;

@Controller
public class HomeController {
	@GetMapping("/")
	public String index(Model model) {
		model.addAttribute("operator", "+");
		model.addAttribute("view", "views/calculatorForm");
		return "base-layout";
	}

	@PostMapping("/")
	public String calculate (@RequestParam String leftOperand,
							 @RequestParam String operator,
							 @RequestParam String rightOperand,
							 Model model) {

		double leftNumber;
		double rightNumber;

		try {
			leftNumber = Double.parseDouble(leftOperand);
		} catch (NumberFormatException ex) {
			leftNumber = 0;
		}
		try {
			rightNumber = Double.parseDouble(rightOperand);
		}catch (NumberFormatException ex) {
			rightNumber = 0;
		}

		Calculator calculator = new Calculator(leftNumber, operator, rightNumber);

		double result = calculator.calculateResult();

		model.addAttribute("leftOperand", calculator.getLeftOperand());
		model.addAttribute("operator", calculator.getOperator());
		model.addAttribute("rightOperand", calculator.getRightOperand());

		model.addAttribute("result", result);


		model.addAttribute("view", "views/calculatorForm");

		return "base-layout";
	}

}

File: Tech Module/Software Technologies/C#/Calculator/Calculator-CSharp/Controllers/HomeController.cs
using System.Web.Mvc;
using System.Web;
using Calculator_CSharp.Models;
using System;

namespace Calculator_CSharp.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index(Calculator calculator)
        {
            return View(calculator);
        }

        [HttpPost]
        public ActionResult Calculate (Calculator calculator)
        {
            calculator.Result = calculateResult(calculator);

            return RedirectToAction("Index", calculator);
        }

        public decimal calculateResult(Calculator calculator)
        {
            decimal result = 0;

            if (calculator.Operator == "+")
            {
                result = calculator.LeftOperand + calculator.RightOperand;
            }
            else if (calculator.Operator == "-")
            {
                result = calculator.LeftOperand - calculator.RightOperand;
            }
            else if (calculator.Operator == "/")
            {
                if (calculator.RightOperand != 0)
                {
                    result = calculator.LeftOperand / calculator.RightOperand;
                }

            }
            else if (calculator.Operator == "*")
            {
                result = calculator.LeftOperand * calculator.RightOperand;
            }
            // Calculate the power of given base with given exponent!
            // The base is LeftOperand and exponent is the RightOperand!
            else if (calculator.Operator == "^")    
            {                                    
                result = (decimal)(Math.Pow((double)calculator.LeftOperand, (double)calculator.RightOperand));
            }
            // Calculate the N-th root of given base!
            // RightOperand is base and LeftOperand is the given root!
            else if (calculator.Operator == '\u221A'.ToString())    
            {                                       
                if (calculator.LeftOperand != 0)
                {
                    result = (decimal)(Math.Pow((double)calculator.RightOperand, (double)(1 / calculator.LeftOperand)));
                }
            }


            return result;
        }
    }
}